ui/src/App.jsx

```diff
@@ -13,18 +15,20 @@ import {
   serverToUIConversion
 } from "./utils/stateConversion";
 
-class App extends Component {
-  constructor(props) {
-    super(props);
-    this.sendUpdate = this.sendUpdate.bind(this);
-    this.refreshPolicy = this.refreshPolicy.bind(this);
-  }
```

Comment:
Hmm I have come across articles online that state that `flow` by itself is good enough to replace the weaker `prop-types`:
- https://www.reddit.com/r/reactjs/comments/9zcfk7/should_i_use_proptypes_or_flow_in_my_react/
- https://gaya.pizza/articles/using-flow-instead-of-prop-types/

Argument is `flow` already augmented the type checking to be at a level better than `prop-types` anyway, and can be done at compile-time instead of runtime.

Reply:
Also, is there a reason for not using the newer ES6 `class`/`constructor` syntax here?

Reply:
https://overreacted.io/how-are-function-components-different-from-classes/

Reply:
:+1: Good read to learn about the advantage of React function over ES6 class.

I think in this case, React hooks / functions avoided doing closure capturing + mutability issue, by trying to go with closure capture + immutablility instead to prevent headaches.

I suspect closure capturing + mutability is a very common issue in many other languages, especially those languages that cannot enforce immutability of vars at compile-time + not explicit when capturing vars, making it hard to tell if the `x.y.z` value is copied or not when captured.